opm/parser/eclipse/EclipseState/Grid/FaultCollection.cpp

```diff
@@ -33,42 +37,51 @@
 
 namespace Opm {
 
+
     FaultCollection::FaultCollection()
     {}
 
-
-    FaultCollection::FaultCollection( std::shared_ptr<const GRIDSection> deck, std::shared_ptr<const EclipseGrid> grid) {
-        const auto& faultKeywords = deck->getKeywordList<ParserKeywords::FAULTS>();
+    FaultCollection::FaultCollection(const GRIDSection& gridSection,
+                                     const EclipseGrid& grid) {
+        const auto& faultKeywords = gridSection.getKeywordList<ParserKeywords::FAULTS>();
 
         for (auto keyword_iter = faultKeywords.begin(); keyword_iter != faultKeywords.end(); ++keyword_iter) {
             const auto& faultsKeyword = *keyword_iter;
             for (auto iter = faultsKeyword->begin(); iter != faultsKeyword->end(); ++iter) {
                 const auto& faultRecord = *iter;
                 const std::string& faultName = faultRecord.getItem(0).get< std::string >(0);
-                int I1 = faultRecord.getItem(1).get< int >(0) - 1;
-                int I2 = faultRecord.getItem(2).get< int >(0) - 1;
-                int J1 = faultRecord.getItem(3).get< int >(0) - 1;
-                int J2 = faultRecord.getItem(4).get< int >(0) - 1;
-                int K1 = faultRecord.getItem(5).get< int >(0) - 1;
-                int K2 = faultRecord.getItem(6).get< int >(0) - 1;
-                FaceDir::DirEnum faceDir = FaceDir::FromString( faultRecord.getItem(7).get< std::string >(0) );
-                std::shared_ptr<const FaultFace> face = std::make_shared<const FaultFace>(grid->getNX() , grid->getNY() , grid->getNZ(),
-                                                                                          static_cast<size_t>(I1) , static_cast<size_t>(I2) ,
-                                                                                          static_cast<size_t>(J1) , static_cast<size_t>(J2) ,
-                                                                                          static_cast<size_t>(K1) , static_cast<size_t>(K2) ,
-                                                                                          faceDir);
-                if (!hasFault(faultName)) {
-                    addFault( faultName );
-                }
-
-                {
-                    Fault& fault = getFault( faultName );
-                    fault.addFace( face );
-                }
+
+                initFault(grid, faultRecord, faultName);
```

Comment:
This will typically be called repeatedly - i.e. it is not a typical `init` function. A better name would be `addFaultFaces()`